### PR TITLE
feat: fetch current/next assignments of a sharehouse

### LINKED ISSUE #251: Fetch current/next assignments of a sharehouse
## Description

Please fetch data from `/api/rotation/current/:share_house_id` or `/api/rotation/next/:share_house_id` to show current/next assignments of a sharehouse.

## Criteria

- [ ] Fetch and display current assignment data from `/api/rotation/current/:share_house_id`
- [ ] Fetch and display next assignment data from `/api/rotation/next/:share_house_id`
- [ ] If applicable, delete hard-coded code like mock data or related code

### PR DESCRIPTION
**Pull request by KaiKoide**
## Overview

Fetch data from `/api/rotation/:share_house_id` and display the data on the dashboard page of each share house.

## Changes

- create `/api/rotation/:share_house_id` endpoint to get data of current and next assignment

## Review points

- display current assignment data from /`api/rotation/:share_house_id` on current tab
- display next assignment data from `/api/rotation/:share_house_id` on next tab
- add query parameters to recognize the current active tab

## Screenshots or videos

**current tab**
image

**next tab**
image

## Assignee Checklist:

- [ ] The base branch is correct (no accidental merges)
- [ ] The branch name follows our branch naming rules
- [ ] The PR title follows our PR title rules
- [ ] My code follows our coding style
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings or errors in the console

## Reviewer Checklist:

- [x] Code readability and simplicity
- [x] Follows best practices and coding standards
- [x] Understandable and maintainable code

### CODE REVIEW

File: src/app/sharehouses/[share_house_id]/page.tsx [hunk 1]
```diff
@@ -17,119 +19,58 @@ export interface IShareHousePageProps {
...
-      tenant: 'Akio',
-      taskNum: 9,
-    },
-    {
-      category: 'Living room',
-      tenant: 'Maaaatio',
-      taskNum: 7,
-    },
-  ];
+  const formatDate = (date: string) => {
```
Comment (akiomatic) [inline src/app/sharehouses/[share_house_id]/page.tsx]: Could you use this function to formate dates?
https://github.com/Tascurator/tascurator-frontend/blob/5273ed95d352a327f50bbff950b212b75db3699a/src/utils/dates.ts#L90

Comment (akiomatic) [reply src/app/sharehouses/[share_house_id]/page.tsx]: Also, could you convert the date to PDT?

**Review by akiomatic — CHANGES_REQUESTED** [review]
Thank you. Could you check my comments?

File: src/app/sharehouses/[share_house_id]/page.tsx (hunk 1 above)
Comment (KaiKoide) [reply src/app/sharehouses/[share_house_id]/page.tsx]: I imported the `formatDate` method and converted the date to PDT.

**Review by akiomatic — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**